### ISSUE TITLE
Allow for symbolic properties

### ISSUE THREAD

**Issue by bkuster** [openlayers/openlayers#7757]
I've been using `Symbol` more and more often in my code to track _private_ properties. I've been trying to attach such properties to `ol.Feature`, at first simply as static properties, but now I wish to add some _watched_ symbolic properties. Sadly this line: https://github.com/openlayers/openlayers/blob/72ca7b28c6cfcff3b7b2e218bcc649cc1740b321/src/ol/object.js#L91
prevents this from working. I replaced `key` with  `key.toString()` in this line to allow for the key to be a `Symbol` and its been working fine so far.

I can't shake the feeling that I'm missing something. Is there a reason for not allowing symbolic properties? Is this something other people might also be interested in? If so, I'll take the time to edit the docs too and open a PR.

**Comment by raiyni** [comment]
Because Symbol is ES6. 

You can just store them on the raw object if you want private access and dispatch your own event.

**Comment by bkuster** [comment]
Dear Ron,

thanks for the comment, but I assume you miss understood my request, so let me rephrase: I _will_ be using `Symbol` in my code and I have been adapting openlayers to accommodate them. Writing a custom `Observer` class to wrap around an already well tested and established system of observables and events just isn't practical or solution oriented.

I am fully aware that `Symbol` is part of the newest standard of JavaScript. None the less, I assume openlayers is a library that should find application in newer JS code as well. And if changing a few lines of code is all it takes to allow symbolic properties, I don't see the reason for rejecting the entire subject simply because it references a newer standard of JS (I'm not saying openlayers should _use_ `Symbol` in its code).

I've opened this issue to ask the openlayers community: Has anyone else been dealing with this issue or has an interest in using `Symbol` and whether I should make my adaptions available in a PR.

Best, Ben

**Comment by marcjansen** [comment]
I think a PR as outlined above is totally acceptable.

**Comment by raiyni** [comment]
> I can't shake the feeling that I'm missing something. Is there a reason for not allowing symbolic properties? 

I gave you the reason. OpenLayers wasn't created in ES6 time and only very recently is moving to ES modules. 

> Writing a custom Observer class to wrap around an already well tested and established system of observables and events just isn't practical or solution oriented.

You wouldn't have to write a custom Observer class...

```ol.Feature.prototype.setSymbol = function(key, value) {
  
  this[key] = value;
  
  this.dispatchEvent('symbolChange')
}
```

Later.

```feature.on('symbolChange', function(e) {
  console.log(e)
})


madrid.setSymbol('abc', 'sdf')
```

Regardless if it gets pushed into the library or not.

**Comment by bkuster** [comment]
@raiyni thanks, I hadn't thought of extending the feature, I'll keep that in mind, it might help me with some other things I need to figure out.

I'll open a PR for this nonetheless, but with:
> only very recently is moving to ES modules

I can wait for this to go through. I haven't read up on openlayers v5 changes yet, but if it includes ES6 support for the **specs**, I could then extend the specs to cover this as well (or is this already possible with the closure compiler? Haven't found the line where the input language is set to `ES6`)
